core/drivers/crypto/se050/session.c

```diff
@@ -0,0 +1,84 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (C) Foundries Ltd. 2020 - All Rights Reserved
+ * Author: Jorge Ramirez <jorge@foundries.io>
+ */
+
+#include <config.h>
+#include <initcall.h>
+#include <se050.h>
+
+sss_se05x_key_store_t *se050_kstore;
+sss_se05x_session_t *se050_session;
+struct sss_se05x_ctx se050_ctx;
+
+TEE_Result se050_core_early_init(struct se050_scp_key *keys)
+{
+	sss_status_t status = kStatus_SSS_Success;
+
+	status = se050_session_open(&se050_ctx, keys);
+	if (status != kStatus_SSS_Success)
+		return TEE_ERROR_GENERIC;
+
+	if (IS_ENABLED(CFG_CORE_SE05X_INIT_NVM)) {
+		status = se050_factory_reset(&se050_ctx.session.s_ctx);
+		if (status != kStatus_SSS_Success)
+			return TEE_ERROR_GENERIC;
+	}
+
+	if (se050_ctx.session.subsystem == kType_SSS_SubSystem_NONE)
+		return TEE_ERROR_GENERIC;
+
+	status = se050_key_store_and_object_init(&se050_ctx);
+	if (status != kStatus_SSS_Success)
+		return TEE_ERROR_GENERIC;
+
+	se050_session = (sss_se05x_session_t *)((void *)&se050_ctx.session);
+	se050_kstore = (sss_se05x_key_store_t *)((void *)&se050_ctx.ks);
+
+	return TEE_SUCCESS;
+}
+
+static TEE_Result display_info(void)
+{
+	se050_display_board_info(se050_session);
+	sss_se05x_session_close(se050_session);
```

Comment:
why closing and initializing again ?

Reply:
the se050 runs JCOP3  (java card open platform); buried in their software stuch it is mentioned that accessing the board information requires access to the ISD (Issuer Security Domain). And that once that has been accessed, we cant go back to the Java Card Secure Element domain ( Applet) so we have to close and reopen the session.  I did verify that this is correct (ie, unless we close/reopen the applet does not respond).

Reply:
Thank for the explanation. Obviosuly needed :) I suggest you add an inline comment to briefly state that, something short like `/* SE050 ISD access mandates reopening session for further exchanges */`